### PR TITLE
chore(prometheus source): Update hyper

### PR DESCRIPTION
**Pull request by ktff**
Ref. #2117

This also updates `hyper-openssl` in the source.

### CODE REVIEW

File: src/hyper.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use bytes::{Bytes, BytesMut};
+use futures::TryStreamExt;
+use hyper13::{Body, Error};
+
+// Can be eliminated once we start using `bytes` crate with the same version as `hyper`.
```
Comment (binarylogic) [inline src/hyper.rs:5]: Mind prefixing this with `TODO: `? @Hoverbear @lukesteensen how are we handling these types of tasks?

Comment (binarylogic) [reply src/hyper.rs:5]: I'm also realizing this is a carry-over from the older code.

Comment (ktff) [reply src/hyper.rs:5]: A `TODO` would be helpful.

**Comment by ktff** [comment]
Noting, not to merge this, a few tests have started failing here semi regularly.

**Comment by binarylogic** [comment]
The "flaky" tests or others?

**Comment by ktff** [comment]
The "flaky", but they were failing for a good reason, I accidentally removed 
feature requirement `#[cfg(feature = "aws-ec2-metadata-integration-tests")]` 
while moving the above mentioned code.

File: src/hyper.rs (hunk 1 above)
Comment (ktff) [reply src/hyper.rs]: I'll added since it can't hurt.

**Comment by ktff** [comment]
Fixed. 

@binarylogic should we wait for @Hoverbear or go with merger?

**Comment by Hoverbear** [comment]
I can review :)

File: src/sources/prometheus/mod.rs [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
     topology::config::GlobalOptions,
     Event,
 };
-use futures01::{sync::mpsc, Future, Sink, Stream};
+use futures::{FutureExt, TryFutureExt};
+use futures01::{sync::mpsc, Future, Sink, Stream as Stream01};
 use http::Uri;
-use hyper_openssl::HttpsConnector;
+use hyper13::{Body, Client, Request};
+use hyper_openssl08::HttpsConnector;
```
Comment (Hoverbear) [inline src/sources/prometheus/mod.rs:13]: Gosh it would be so nice to remove some of these versioned deps. :(

Comment (ktff) [reply src/sources/prometheus/mod.rs:13]: They will, this and one more PR, and they are out of here. At least those.

**Comment by fanatid** [comment]
@ktff I found that hyper:0.12.35 still used in `mod tests` in prometheus source. Can you update it too?

**Pull request merged**